### PR TITLE
Add websocket support and fix port override

### PR DESCRIPTION
**Pull request by haslinghuis**
- Thanks to @AJ92 to get this started using provided websockets client code, had to fix port override, serial shim and serial backend to get it started.

**Requirements**
- Needs implementation of websockets in firmware likely https://libwebsockets.org/
- Alternative socket servers: https://github.com/Theldus/wsServer or https://github.com/bsapundzhiev/ucwebsocket

**Additional info**
- Web Socket Protocol: https://datatracker.ietf.org/doc/html/rfc6455
- Ultimate sockets guide: https://beej.us/guide/bgnet/html/
- https://news.ycombinator.com/item?id=32867644

**Alternative protocol**
- webtransport: https://ietf-wg-webtrans.github.io/draft-ietf-webtrans-http3/draft-ietf-webtrans-http3.html

Screenshot from 2024-09-24 18-14-28

### CODE REVIEW

File: src/js/protocols/websocket.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    waitForConnection(socket) {
+        return new Promise((resolve) => {
+            const interval = setInterval(() => {
+                if (socket.connected) {
+                    clearInterval(interval);  // Stop checking
+                    resolve();                // Resolve the promise
+                }
+            }, 100); // Check every 100ms, adjust as needed
+        });
+    }
```
Comment (AJ92) [inline src/js/protocols/websocket.js:69]: works without this too

File: src/js/protocols/websocket.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            console.log(`${socket.logHead} Connected: `, e);
+            socket.connected = true;
+            socket.dispatchEvent(
+                new CustomEvent("connect", { detail: {
+                    socketId: socket.address,
+                }}),
+            );
+        };
+
+        await this.waitForConnection(socket);
```
Comment (AJ92) [inline src/js/protocols/websocket.js:88]: not needed unless 1:1 functionality compared with webSerial is desired

**Review by blckmn — APPROVED** [review]

**Comment by nerdCopter** [comment]
how to test? i was able to flash FC.

**Comment by AJ92** [comment]
> how to test? i was able to flash FC.

not related to hw

sitl target needs to be adapted first, if that hasnt been done yet

**Review by nerdCopter — APPROVED** [review]
- approving for workflow; i have no clue on this stuff

**Comment by haslinghuis** [comment]
- Fixes Vue port override.
- This PR adds provisional support for websockets (tested with @AJ92 simulator code).
- Manual mode is broken in PWA without firmware support for websockets or perhaps webtransport.
- Issue is LWS uses cmake instead of make, so have no clue about integrating the library in firmware.

**Pull request merged**